File: app/src/main/java/com/ebsoft/watchlist/network/Yahoo/YahooAPI.java
package com.ebsoft.watchlist.network.Yahoo;

import com.ebsoft.watchlist.data.model.Yahoo.SymbolSearch;

import io.reactivex.Observable;
import retrofit2.Response;
import retrofit2.http.GET;
import retrofit2.http.Query;

/**
 * Created by endre on 09/09/18.
 */

public interface YahooAPI {

    @GET("autoc?format=json&region=1&lang=en")
    Observable<Response<SymbolSearch>> searchSymbol(@Query("query") String symbol);
}

File: app/src/main/java/com/ebsoft/watchlist/data/control/network/Yahoo/YahooAPI.kt
package com.ebsoft.watchlist.data.control.network.Yahoo

import com.ebsoft.watchlist.data.model.yahoo.SymbolSearch

import io.reactivex.Observable
import kotlinx.coroutines.Deferred
import retrofit2.Response
import retrofit2.http.GET
import retrofit2.http.Query

/**
 * Created by endre on 09/09/18.
 */

interface YahooAPI {

    @GET("autoc?format=json&region=1&lang=en")
    fun searchSymbol(@Query("query") symbol: String): Deferred<Response<SymbolSearch>>
}